bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java

```diff
@@ -129,6 +131,22 @@ boolean complete(BookieSocketAddress host, final ByteBuf buffer) {
         }
 
         /**
+         * Fail the request with given result code <i>rc</i>.
+         *
+         * @param rc
+         *          result code to fail the request.
+         * @return true if we managed to fail the entry; otherwise return false if it already failed or completed.
```

Comment:
what will be case - when the control is here (i.e fail) the read is completed but not failed? If so may be we need more explanation on that scenario. May be two parallel reads, one fail and other success, ?